src/zenml/orchestrators/step_launcher.py

```diff
@@ -212,6 +213,7 @@ def launch(self) -> None:
         Raises:
             RunStoppedException: If the pipeline run is stopped by the user.
         """
+        publish_utils.step_exception_info.set(None)
```

Comment:
Is there any way this `launch`method is called by two parallel running processes? As we discussed yesterday, this might create a situation where they overwrite each other.

Reply:
Reminds me of [this](https://github.com/zenml-io/zenml/pull/3812#discussion_r2215597418) as well.